baremetal: Use runtimecfg for Corefile rendering

celebdor opened the pull request.

bcrochet committed:
baremetal: Add coredns, mdns-publisher, haproxy, keepalived static pods

Adds pods to master and worker nodes as appropriate

Updates haproxy container to use openshift/router-haproxy image instead of
docker.io/library/haproxy

Also adds liveness tests for the coredns,mdns-publisher,
haproxy and keepalived static pods, changes worker node
/etc/resolv.conf to point to node's IP instead of 127.0.0.1 and
fix the bug generating haproxy cfg file

Depends-On: #943
Depends-On: #984

Conflicts:
	templates/common/baremetal/files/baremetal-coredns-corefile.yaml
	templates/common/baremetal/files/baremetal-coredns.yaml
	templates/common/baremetal/files/baremetal-mdns-publisher.yaml
	templates/master/00-master/baremetal/files/baremetal-haproxy-haproxy.yaml
	templates/master/00-master/baremetal/files/baremetal-haproxy.yaml
	templates/master/00-master/baremetal/files/baremetal-keepalived-keepalived.yaml
	templates/master/00-master/baremetal/files/baremetal-mdns-config.yaml
	templates/master/00-master/baremetal/files/dhcp-dhclient-conf.yaml
	templates/worker/00-worker/baremetal/files/baremetal-mdns-config.yaml
	templates/worker/00-worker/baremetal/files/dhcp-dhclient-conf.yaml

celebdor committed:
baremetal: Use runtimecfg for Corefile rendering

Signed-off-by: Antoni Segura Puimedon <antoni@redhat.com>

celebdor committed:
baremetal: bootstrap keepalived use runtimecfg

Signed-off-by: Antoni Segura Puimedon <antoni@redhat.com>

bcrochet commented on manifests/baremetal/coredns-corefile.tmpl:
```diff
@@ -1,12 +1,12 @@
     health
-    mdns $DOMAIN $NUM_DNS_MEMBERS $NAME
-    forward . /etc/coredns/resolv.conf
+    mdns {{ .Infra.Status.EtcdDiscoveryDomain }} {{`{{.Cluster.MasterAmount}}`}} {{`{{.Cluster.Name}}`}}
```
Why .Infra.Status? You should be able to just write {{ .EtcdDiscoveryDomain }}.

cybertron committed:
Fixes for templated config rendering

There were a few things needed to make the previous commits fully
functional. First, the images were not wired in to the bootstrap
manifest rendering. Second, the templates themselves had a few
issues that needed to be resolved (capitalization, indentation, etc.).

This passes the necessary image details into the bootstrap manifest
rendering code and cleans up the templates. After this patch the
bootstrap services can be started successfully.

bcrochet committed:
Refer to the correct template vars

Infra is already in ControllerConfig. We can use that. Also,
EtcdDiscoveryDomain is also in ControllerConfig.

This rebases on top of current master that includes the
Infra patch from Roy.

celebdor committed:
bootstrap: ONly render baremetal if needed

Signed-off-by: Antoni Segura Puimedon <antoni@redhat.com>

celebdor committed:
baremetal: update bindata

Signed-off-by: Antoni Segura Puimedon <antoni@redhat.com>

celebdor committed:
baremetal: fix non-virtual-ip for py3.6+

Signed-off-by: Antoni Segura Puimedon <antoni@redhat.com>
